test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -1,230 +1,732 @@
+            New-Item -Path $testFile2 -ItemType File -Value $testContent2 > $null
+        }
+
+        AfterEach {
+            Remove-Item -Path * -Recurse -Force -ErrorAction SilentlyContinue
+        }
+
+        It "Validate Array Input for Path" {
+            $result = Get-Content -Path @($testFile,$testFile2)
+            $result[0] | Should Be $testContent
+            $result[1] | Should Be $testContent2
+        }
+
+        It "Validate Include" {
+            $result = Get-Content -Path "TestDrive:\*" -Include "*2.txt"
+            $result | Should Be $testContent2
+        }
+
+        It "Validate Exclude" {
+            $result = Get-Content -Path "TestDrive:\*" -Exclude "*2.txt"
+            $result | Should Be $testContent
+        }
+
+        It "Validate Filter" {
+            $result = Get-Content -Path "TestDrive:\*" -Filter "*2.txt"
+            $result | Should Be $testContent2
+        }
+
+        It "Validate ReadCount" {
+            Set-Content -Path $testFile -Value "Test Line 1`nTest Line 2`nTest Line 3`nTest Line 4`nTest Line 5`nTest Line 6"
+            $result = (Get-Content -Path $testFile -ReadCount 2)
+            $result[0][0] | Should Be "Test Line 1"
+            $result[0][1] | Should Be "Test Line 2"
+            $result[1][0] | Should Be "Test Line 3"
+            $result[1][1] | Should Be "Test Line 4"
+            $result[2][0] | Should Be "Test Line 5"
+            $result[2][1] | Should Be "Test Line 6"
+        }
+
+        It "Validate TotalCount" {
+            Set-Content -Path $testFile -Value "Test Line 1`nTest Line 2`nTest Line 3`nTest Line 4`nTest Line 5`nTest Line 6"
+            $result = Get-Content -Path $testFile -TotalCount 4
+            $result[0] | Should Be "Test Line 1"
+            $result[1] | Should Be "Test Line 2"
+            $result[2] | Should Be "Test Line 3"
+            $result[3] | Should Be "Test Line 4"
+            $result[4] | Should BeNullOrEmpty
+        }
+
+        It "Validate Tail" {
+            Set-Content -Path $testFile -Value "Test Line 1`nTest Line 2`nTest Line 3`nTest Line 4`nTest Line 5`nTest Line 6"
+            $result = Get-Content -Path $testFile -Tail 2
+            $result[0] | Should Be "Test Line 5"
+            $result[1] | Should Be "Test Line 6"
+            $result[2] | Should BeNullOrEmpty
+        }
     }
 
-    It "Verify HOME" {
-        $homePath = $HOME
-        $tildePath = (resolve-path -Path ~).Path
-        $homePath | Should Be $tildePath
+    Context "Validate Resolve-Path Cmdlet Parameters" {
+        It "Verify LiteralPath" {
+            $result = Resolve-Path -LiteralPath "TestDrive:\"
+            ($result.Path.TrimEnd('/\')) | Should Be "TestDrive:"
+        }
     }
 
-    It "Verify relative" {
-        $relativePath = Resolve-Path -Path . -Relative
-        $relativePath | Should Be (Join-Path "." "")
+    Context "Validate Join-Path Cmdlet Parameters Remotely" {
+        BeforeAll {
```

Comment:
Cleanup is not necessary because TestDrive:\ gets cleaned up at the end of every Context.